### PR TITLE
[BEAM-192] [BEAM-218] Revises main website landing page and adds documentation landing page

### PR DESCRIPTION
**Pull request by devin-donnelly**
This is a preliminary change to the main website landing page and assorted other landing pages. Notable changes include:
- Reorganizes the main landing page and improves the language of the Beam explanation
- Removes some copy-pasted verbiage from Dataflow that was too wordy
- Adds status tables for Beam SDKs and Beam Runners
- Adds a /docs/ subdirectory for documentation
- Adds a landing page under /docs/ with some preliminary docs material
- Adds a coming-soon.html as a placeholder for content not yet available, preventing 404s

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -11,26 +11,67 @@ layout: default
 
 # Apache Beam (incubating)
 
-Apache Beam is an open source, unified model and set of language-specific SDKs for defining data processing workflows that may then be executed on top of a set of supported runners, currently including [Apache Flink](http://flink.apache.org), [Apache Spark](http://spark.apache.org), and [Google Cloud Dataflow](https://cloud.google.com/dataflow).
+Apache Beam is an open source, unified programming model that you can use to define data processing workflows. You define a workflow, called a **pipeline**, by building a program with a Beam SDK. The pipeline is then executed by one of Beam's supported **distributed processing back-ends**, which include [Apache Flink](http://flink.apache.org), [Apache Spark](http://spark.apache.org), and [Google Cloud Dataflow](https://cloud.google.com/dataflow).
```
Comment (davorbonaci) [inline index.md]: Do we need a distrinction between workflow and pipeline? How about the following?

Apache Beam is an open source, unified programming model that you can use to define data processing pipelines. You start by build a pipeline with one of the Beam SDKs, and then execute that pipeline on one of Beam's supported **distributed processing back-ends**, which include...

File: index.md [hunk 2]
```diff
@@ -11,26 +11,67 @@ layout: default
...
 
-In particular, Beam pipelines can represent high-volume computations, where the steps in your job need to process an amount of data that exceeds the memory capacity of a cost-effective cluster. Beam is particularly useful for [Embarrassingly Parallel](http://en.wikipedia.org/wiki/Embarassingly_parallel) data processing tasks, in which the problem can be decomposed into many smaller bundles of data that can be processed independently and in parallel.
+## Apache Beam SDKs
 
-You can also use Beam for Extract, Transform, and Load (ETL) tasks and pure data integration. These tasks are useful for moving data between different storage media and data sources, transforming data into a more desirable format, or loading data onto a new system.
+The Beam SDKs provide a unified data model that can represent and transform any size data set, whether the input is a finite data set from a batch data source, or an infinite data set from a streaming data source. The Beam SDKs use the same classes to represent both bounded and unbounded data, and the same transforms to operate on that data. You use the Beam SDK of your choice to build a program that defines your data processing workflow:
```
Comment (davorbonaci) [inline index.md]: data model -> programming model?

File: index.md (hunk 1 above)
Comment (devin-donnelly) [inline index.md]: The original wording said "workflow", which I assumed is because big data processing audiences may be more familiar with the term "workflow" while "pipeline" is more of a Beam term-of-art. I was thinking of the "define at first use" theory on communicating a new term.

File: index.md [hunk 3]
```diff
@@ -11,26 +11,67 @@ layout: default
...
 
-## Programming Model
-Beam provides a simple and [elegant programming model](https://cloud.google.com/dataflow/model/programming-model) to express your data processing jobs. Each job is represented by a data processing pipeline that you create by writing a program with Beam. Each pipeline is an independent entity that reads some input data, performs some transforms on that data to gain useful or actionable intelligence about it, and produces some resulting output data. A pipeline’s transform might include filtering, grouping, comparing, or joining data.
+Beam currently supports SDKs in the following languages:
```
Comment (davorbonaci) [inline index.md]: Beam currently supports the following language-specific SDKs.

(there could be multiple SDKs in the same language one day)

File: index.md [hunk 4]
```diff
@@ -11,26 +11,67 @@ layout: default
...
-* **Powerful data transforms.** Beam provides several core data transforms that you can apply to your data. These transforms, called PTransforms, are generic frameworks that apply functions that you provide across an entire data set.
-* **I/O APIs for a variety of data formats.** Beam provides APIs that let your pipeline read and write data to and from a variety of formats and storage technologies. Your pipeline can read text files, Avro files, and more.
+## Apache Beam Pipeline Runners
 
-See the [programming model documentation](https://cloud.google.com/dataflow/model/programming-model) to learn more about how Beam implements these concepts.
+The Beam Pipeline Runners translate the data processing workflow you define with your Beam program into the API compatible with the distributed processing back-end of your choice. When you run your Beam program, you'll need to specify the appropriate runner for the back-end service where you want to execute your workflow.
+
+Beam currently supports Runners that work with the following distributed processing back-ends:
```
Comment (davorbonaci) [inline index.md]: A side note about local execution?

**Comment by davorbonaci** [comment]
Looks great! Left a few minor comments. Feel free to accept / reject as you see fit.

It would be great to stage changes somewhere, so we can see it in the presentation view in a browser.

File: index.md (hunk 1 above)
Comment (davorbonaci) [inline index.md]: Based on Google search, "data processing pipeline" seems more popular than "data processing workflow", so not really sure that "workflow" is more widely known. It could be a circular definition ;)  but, I see your logic too.

Comment (devin-donnelly) [inline index.md]: Yep, I'm not sure my reasoning is accurate. I'd like to get James' opinion from the product side.

File: index.md [hunk 5]
```diff
@@ -11,26 +11,67 @@ layout: default
...
+
+<table class="table table-condensed">
+<tr>
+  <th>Runner</th>
+  <th>Status</th>
+</tr>
+<tr class="success">
+  <td>Google Cloud Dataflow</td>
+  <td>Generally Available</td>
```
Comment (evilsoapbox) [inline index.md]: I think someone reading this may assume the runner is GA and ready for primetime use. Should we really say this is also in development? @davorbonaci can correct me if the runner is really GA and ready for everyday use.

File: index.md [hunk 6]
```diff
@@ -11,26 +11,67 @@ layout: default
...
-Beam provides a simple and [elegant programming model](https://cloud.google.com/dataflow/model/programming-model) to express your data processing jobs. Each job is represented by a data processing pipeline that you create by writing a program with Beam. Each pipeline is an independent entity that reads some input data, performs some transforms on that data to gain useful or actionable intelligence about it, and produces some resulting output data. A pipeline’s transform might include filtering, grouping, comparing, or joining data.
+Beam currently supports SDKs in the following languages:
 
-Beam provides several useful abstractions that allow you to think about your data processing pipeline in a simple, logical way. Beam simplifies the mechanics of large-scale parallel data processing, freeing you from the need to manage orchestration details such as partitioning your data and coordinating individual workers.
+<table class="table table-condensed">
+<tr>
+  <th>Language</th>
+  <th>SDK Status</th>
+</tr>
+<tr class="success">
```
Comment (evilsoapbox) [inline index.md]: This is a personal (and 100% subjective) nit but the green background is sort of distracting on the page overall. What about using icons or glyphs in the table to indicate status? That may also be more accessible overall.

File: index.md (hunk 5 above)
Comment (davorbonaci) [inline index.md]: @evilsoapbox is right. (Beam hasn't had any releases yet, so nothing is "available" yet.)

File: index.md (hunk 1 above)
Comment (evilsoapbox) [inline index.md]: +1 on using "pipeline" alone.

File: index.md (hunk 2 above)
Comment (evilsoapbox) [inline index.md]: +1

File: index.md (hunk 6 above)
Comment (devin-donnelly) [inline index.md]: There really isn't anything to differentiate right now, so we can come up with a way to do it when we have something to differentiate.

File: index.md (hunk 5 above)
Comment (devin-donnelly) [inline index.md]: Changed.

File: index.md (hunk 4 above)
Comment (devin-donnelly) [inline index.md]: Added at the bottom.

File: index.md (hunk 3 above)
Comment (devin-donnelly) [inline index.md]: Yep, done.

File: index.md (hunk 2 above)
Comment (devin-donnelly) [inline index.md]: Done, modulo "data sets of any size" (you aren't limited to one data set per pipeline).

File: index.md (hunk 1 above)
Comment (devin-donnelly) [inline index.md]: Changed to something close to Davor's suggestion, and changed all other instances of "workflow" to "pipeline" as necessary.

File: index.md [hunk 7]
```diff
@@ -11,26 +11,69 @@ layout: default
 
 # Apache Beam (incubating)
 
-Apache Beam is an open source, unified model and set of language-specific SDKs for defining data processing workflows that may then be executed on top of a set of supported runners, currently including [Apache Flink](http://flink.apache.org), [Apache Spark](http://spark.apache.org), and [Google Cloud Dataflow](https://cloud.google.com/dataflow).
+Apache Beam is an open source, unified programming model that you can use to create a data processing **pipeline**. You start by building a program that defines the pipeline using one of the open source Beam SDKs. The pipeline is then executed by one of Beam's supported **distributed processing back-ends**, which include [Apache Flink](http://flink.apache.org), [Apache Spark](http://spark.apache.org), and [Google Cloud Dataflow](https://cloud.google.com/dataflow).
```
Comment (davorbonaci) [inline index.md]: I wouldn't say "by a backend". I'd probably say "by a runner, on a backend". Feel free to reject.

File: index.md [hunk 8]
```diff
@@ -11,26 +11,69 @@ layout: default
...
-In particular, Beam pipelines can represent high-volume computations, where the steps in your job need to process an amount of data that exceeds the memory capacity of a cost-effective cluster. Beam is particularly useful for [Embarrassingly Parallel](http://en.wikipedia.org/wiki/Embarassingly_parallel) data processing tasks, in which the problem can be decomposed into many smaller bundles of data that can be processed independently and in parallel.
+## Apache Beam SDKs
 
-You can also use Beam for Extract, Transform, and Load (ETL) tasks and pure data integration. These tasks are useful for moving data between different storage media and data sources, transforming data into a more desirable format, or loading data onto a new system.
+The Beam SDKs provide a unified programming model that can represent and transform data sets of any size, whether the input is a finite data set from a batch data source, or an infinite data set from a streaming data source. The Beam SDKs use the same classes to represent both bounded and unbounded data, and the same transforms to operate on that data. You use the Beam SDK of your choice to build a program that defines your data processing pipeline:
```
Comment (davorbonaci) [inline index.md]: Not sure if you intended to end with a `:`, instead of a `.`.

File: index.md [hunk 9]
```diff
@@ -11,26 +11,69 @@ layout: default
...
-* **Powerful data transforms.** Beam provides several core data transforms that you can apply to your data. These transforms, called PTransforms, are generic frameworks that apply functions that you provide across an entire data set.
-* **I/O APIs for a variety of data formats.** Beam provides APIs that let your pipeline read and write data to and from a variety of formats and storage technologies. Your pipeline can read text files, Avro files, and more.
+## Apache Beam Pipeline Runners
 
-See the [programming model documentation](https://cloud.google.com/dataflow/model/programming-model) to learn more about how Beam implements these concepts.
+The Beam Pipeline Runners translate the data processing pipeline you define with your Beam program into the API compatible with the distributed processing back-end of your choice. When you run your Beam program, you'll need to specify the appropriate runner for the back-end service where you want to execute your pipeline.
```
Comment (davorbonaci) [inline index.md]: Some backends may not be "services". I might just drop word 'service'. Feel free to reject.

File: index.md [hunk 10]
```diff
@@ -11,26 +11,69 @@ layout: default
```
Comment (davorbonaci) [inline index.md]: Have you, by any chance, missed to include a corresponding change to a `index.html` file in a commit?

File: index.md (hunk 7 above)
Comment (devin-donnelly) [inline index.md]: In this case that's inaccurate. The actual work is done by the backend. The Runner translates the pipeline into the API the back-end expects--in Dataflow's case, JSON.

File: index.md (hunk 8 above)
Comment (devin-donnelly) [inline index.md]: Fixed.

File: index.md (hunk 9 above)
Comment (devin-donnelly) [inline index.md]: Done.

**Comment by davorbonaci** [comment]
LGTM, merge in progress.
